common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -135,6 +145,9 @@ public int weigh(File file, ShuffleIndexInformation indexInfo) {
       executors = Maps.newConcurrentMap();
     }
     this.directoryCleaner = directoryCleaner;
+    if (extraKnownManager != null) {
+      knownManagers.add(extraKnownManager);
+    }
```

Comment:
This looks like external shuffle service must support a specified shuffle manager? Except for built-in shuffle managers in `knownManagers`, are we sure any arbitrary shuffle manager can be well supported by external shuffle service?

Reply:
Yes, if user sets their own shuffle manager (spark.shuffle.manager), they need to be aware that their shuffle manager needs to work or co-exist with external shuffle service. Spark framework should let user make that decision instead of only allowing two hardcoded values ("org.apache.spark.shuffle.sort.SortShuffleManager", "org.apache.spark.shuffle.unsafe.UnsafeShuffleManager").

Reply:
> Yes, if user sets their own shuffle manager (spark.shuffle.manager), they need to be aware that their shuffle manager needs to work or co-exist with external shuffle service.

It's only required when users enable external shuffle service, right?


This change looks likes any shuffle manager will be added to `knownManagers`. If so, do we still need `knownManagers`?

Reply:
Yeah, that's right, I was thinking of that as well :) How about let's totally remove knownManagers checking?

Reply:
Sounds good.

Reply:
Cool, let me update the PR, thanks!

Reply:
Updated, totally removed knownManagers this time.